### DISCUSSION TITLE
Converting version 20 .drawio files to version 21 file format from the command line

### DISCUSSION THREAD

**Discussion by sba923** [jgraph/drawio#4192] [Q&A]
I would like to be able to search the (SharePoint / OneDrive for Business) version history for a .drawio file to determine, for instance, when a particular (text) change was made.

I have extracted all the versions of the file to a folder, with names containing the version number.

So far so good.

But the file format changed between version 20 and version 21, so searching for text in version <= 20 doesn't work, and comparing the last version 20 format file version with the first version 21 format file version yields this kind of useless result:

![image](https://github.com/jgraph/drawio/assets/12860484/528d08da-a22a-451e-a6b3-d37788fe761f)

I'm looking for a command-line, scriptable way to convert those pre-version 21 file format files to the latest file format.

Any help would be greatly appreciated.

**Comment by davidjgraph** [comment]
Both formats are valid in both version, just one is compressed and one not. 

https://jgraph.github.io/drawio-tools/tools/convert.html (view source) shows how to encode/decode.

The compressXml configuration option defines whether or not to compress, see https://www.drawio.com/doc/faq/configure-diagram-editor.

**Comment by sba923** [comment]
Thanks a bunch for your quick response! I'm going to look for a way to convert the compressed/base64 encoded data to the XML form I have in other versions of that .drawio file, using PowerShell code so that I can easily "normalize" the "file versions" collection.

Has the default `compressXml` configuration option default value changed... from `true` to `false` at some point? I don't remember ever changing that...
